### PR TITLE
Merge 7.12 to main

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
None

## Summary
Merge changes from 7.12 to main.

7.12.1 tag: https://github.com/elastic/detection-rules/releases/tag/v7.12.1

_Merge not squash_

### CODE REVIEW

File: detection_rules/schemas/v7_12.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class ThresholdCardinality(jsl.Document):
             field = jsl.StringField(required=True)
             value = jsl.IntField(minimum=1, required=True)
 
-        field = jsl.ArrayField(jsl.StringField(required=True, default=""))
+        field = jsl.ArrayField(jsl.StringField(required=False, min_length=1), required=True)
```
Comment (rw-access) [inline detection_rules/schemas/v7_12.py:34]: this is gonna have a merge conflict with #1084.
but that's easy, we can fix it there

Comment (rw-access) [reply detection_rules/schemas/v7_12.py:34]: i think the required is confusing inside of the array. i'm not sure what what even means
```suggestion
        field = jsl.ArrayField(jsl.StringField(min_length=1), required=True)
```

File: rules/aws/credential_access_aws_iam_assume_role_brute_force.toml [hunk 2]
```diff
@@ -47,6 +47,6 @@ name = "Credential Access"
 reference = "https://attack.mitre.org/tactics/TA0006/"
 
 [rule.threshold]
-field = [""]
+field = []
```
Comment (rw-access) [inline rules/aws/credential_access_aws_iam_assume_role_brute_force.toml:50]: i thought there were two affected rules?

**Review by rw-access — APPROVED** [review]
LGTM, just a few Q's/comments

File: rules/aws/credential_access_aws_iam_assume_role_brute_force.toml (hunk 2 above)
Comment (brokensound77) [reply rules/aws/credential_access_aws_iam_assume_role_brute_force.toml:50]: Just checked again and only see this one

**Comment by rw-access** [comment]
Removed unnecessary merge commits, see 791c911b9e46b33f3bf674de1d2ba8197983ae4c

**Pull request closed without merging**